gitea/statefulset.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: gitea
+  namespace: gitea
+  labels:
+    app: gitea
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: gitea
+  template:
+    metadata:
+      labels:
+        app: gitea
+    spec:
+      securityContext:
+        fsGroup: 1000
+      ##  runAsUser: 1000
+      ##  runAsGroup: 1000
+      containers:
+      - name: gitea
+        image: gitea/gitea:latest
+        env:
+          - name: SSH_LISTEN_PORT
+            value: "22"
+          - name: SSH_PORT
+            value: "22"
+          - name: GITEA_APP_INI
+            value: /data/gitea/conf/app.ini
+          - name: GITEA_CUSTOM
+            value: /data/gitea
+          - name: GITEA_WORK_DIR
+            value: /data
+          - name: GITEA_TEMP
+            value: /tmp/gitea
+          - name: TMPDIR
+            value: /tmp/gitea
+        ports:
+        - containerPort: 3000
+          name: gitea
+        - containerPort: 22
+          name: git-ssh
+        # securityContext:
+        #   allowPrivilegeEscalation: false
+        #   capabilities:
+        #     drop:
+        #     - ALL
+        #     add:
+        #     - SYS_CHROOT
+        #   privileged: false
+        #   readOnlyRootFilesystem: true
+        #   runAsUser: 1000
+        #   runAsNonRoot: true
+        #   runAsGroup: 1000
+        volumeMounts:
+        - name: temp
+          mountPath: /tmp
+        - name: data
+          mountPath: /data
+      volumes:
+      - name: temp
+        emptyDir: {}
+      - name: data
+        persistentVolumeClaim:
+          claimName: gitea
+  volumeClaimTemplates:
```

Comment:
@henderiw There would be an expectation that the cluster where this KRM resource is applied would have storage configured and that it would use the default storage class.  
Am curious as to how you tested this in your local environment running KIND cluster?
Thanks

Reply:
I used the default storage class from the kind cluster.

Reply:
> I used the default storage class from the kind cluster.

Thanks, I see that there is a local-path provisioner created as part of KIND cluster. 
For regular K8S clusters storage provisioners needs to be installed

```
kubectl --kubeconfig ~/.kube/mgmt-config get sc
NAME                 PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
standard (default)   rancher.io/local-path   Delete          WaitForFirstConsumer   false                  44d
```

Reply:
yes default storage class is what we use. So nothing demanding